src/protocols/MeterS0.cpp

```diff
@@ -238,31 +238,34 @@ void MeterS0::counter_thread()
 		if (is_blocking) {
 			bool timeout = false;
 			if (_hwif->waitForImpulse( timeout )) {
+				//Event erkannt. Wert des Events ist, aufgrund des prellen reiner Zufall!!
+				//deswegen erst entprellen .....
+				if (!timeout && (_debounce_delay_ms > 0) ){
+					// nanosleep _debounce_delay_ms
+					struct timespec ts;
+					ts.tv_sec = _debounce_delay_ms/1000;
+					ts.tv_nsec = (_debounce_delay_ms%1000)*1e6;
+					struct timespec rem;
+					while ( (-1 == nanosleep(&ts, &rem)) && (errno == EINTR) ) {
+						ts = rem;
+					}
+				}
+				//  ... und dann weiterarbeiten
 				struct timespec temp_ts;
 				clock_gettime(CLOCK_REALTIME, &temp_ts);
 				_ms_last_impulse = timespec_sub_ms(temp_ts, _time_last_ref); // uses atomic operator=
-				if (_hwif_dir && ( _hwif_dir->status()>0 ) )
+				
+				if (_hwif_dir && ( _hwif_dir->status()>0 ) ) { // bei Zweirichtungszaehler Impulse mit logisch 1 negativ werten
 					++_impulses_neg;
```

Comment:
Following your logic there would be a 
```
if (_hwif->status()>0) ++_impulses_neg; 
```
missing here as well. The dir_pin should only indicate the direction.